test/unit/NodeCompiler/CompileNodeToValueTest.php

```diff
@@ -678,12 +683,108 @@ public function magicConstantsInNamespaceProvider(): array
+                        {
+                        }
+
+                        // @TODO test that all methods are covered (use reflection)
+                        public function getSource(): string
+                        {
+                            return $this->next->getSource();
+                        }
+
+                        public function getName(): ?string
+                        {
+                            return $this->next->getName();
+                        }
+
+                        public function getFileName(): ?string
+                        {
+                            return '/non/existing/path/to/sources.php';
+                        }
+
+                        public function isInternal(): bool
+                        {
+                            return $this->next->isInternal();
+                        }
+
+                        public function getExtensionName(): ?string
+                        {
+                            return $this->next->getExtensionName();
+                        }
+
+                        public function isEvaled(): bool
+                        {
+                            return $this->next->isEvaled();
+                        }
+
+                        public function getAliasName(): ?string
+                        {
+                            return $this->next->getAliasName();
+                        }
+                    },
+                    $identifier
+                );
+            }
+
+            /** {@inheritDoc} */
+            public function findReflectionsOfType(
+                Reflector $reflector,
+                LocatedSource $locatedSource,
+                IdentifierType $identifierType,
+            ): array {
+                throw new \BadMethodCallException('Not expected to be called');
+            }
+        };
+
+        $reflector = (new DefaultReflector(new StringSourceLocator(
+            <<<'PHP'
+<?php
+
+const CURRENT_DIR = __DIR__;
+const CURRENT_FILE = __FILE__;
+PHP
+            ,
+            $astLocatorWithFakeFilePath
+        )));
+
+        self::assertSame(
+            '/non/existing/path/to',
+            $reflector
+                ->reflectConstant('CURRENT_DIR')
+                ->getValue()
+        );
+
+        self::assertSame(
+            '/non/existing/path/to/sources.php',
+            $reflector
+                ->reflectConstant('CURRENT_FILE')
+                ->getValue()
+        );
+    }
+
     /**
      * @dataProvider magicConstantsInNamespaceProvider
      */
     public function testMagicConstantsInNamespace(string $constantName, mixed $expectedValue): void
     {
-        $reflector = new DefaultReflector(new SingleFileSourceLocator(__DIR__ . '/../Fixture/MagicConstants.php', $this->astLocator));
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(realpath(__DIR__ . '/../Fixture/MagicConstants.php'), $this->astLocator));
```

Comment:
Minor breakage here, but the file path should still be valid (even if relative), so IMO this is an OK change for 5.3.0